libraries/chain_kv/include/b1/chain_kv/chain_kv.hpp

```diff
@@ -302,28 +305,49 @@ struct database {
-      options.level_compaction_dynamic_level_bytes = true;
-      options.bytes_per_sync                       = 1048576;
-
-      if (threads)
-         options.IncreaseParallelism(*threads);
+            std::optional<int> max_open_files = {}, std::optional<boost::filesystem::path> options_file_name = {}) {
+      rocksdb::DB* p;
 
-      options.OptimizeLevelStyleCompaction(256ull << 20);
+      if (options_file_name) {
+         // load the options file
+         rocksdb::DBOptions loaded_db_opt;
+         std::vector<rocksdb::ColumnFamilyDescriptor> loaded_cf_descs;
+         check(LoadOptionsFromFile(options_file_name->string(), rocksdb::Env::Default(), &loaded_db_opt, &loaded_cf_descs), "database::database: rocksdb::LoadLatestOptions: ");
+         loaded_db_opt.env->SetBackgroundThreads(loaded_db_opt.max_background_jobs, rocksdb::Env::LOW); // low priority background threads pool
+         loaded_db_opt.env->SetBackgroundThreads(1, rocksdb::Env::HIGH); // high priority background threads pool
+
+         // open rocksdb using the loaded options
+         std::vector<rocksdb::ColumnFamilyHandle*> handles;
+         check(rocksdb::DB::Open(loaded_db_opt, db_path, loaded_cf_descs, &handles, &p), "database::database:rocksdb::DB::Open (options files): ");
+      } else {
+         rocksdb::Options options;
+         options.create_if_missing = create_if_missing;
 
-      if (max_open_files)
-         options.max_open_files = *max_open_files;
+         if (threads) {
+            options.IncreaseParallelism(*threads);
+         } else {
+            options.IncreaseParallelism(20);
+         }
+         if (max_open_files) {
+            options.max_open_files = *max_open_files;
+         } else {
+            options.max_open_files = 768;
+         }
 
-      rocksdb::BlockBasedTableOptions table_options;
-      table_options.format_version               = 4;
-      table_options.index_block_restart_interval = 16;
-      options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+         // Those were from RocksDB Performance Tuning Guide for a typical
+         // setting. Applications are encuroage to experiment different settings
+         // and use options file instead.
+         options.compaction_style = rocksdb::kCompactionStyleLevel; // level style compaction
+         options.level0_file_num_compaction_trigger = 10; // number of L0 files to trigger L0 to L1 compaction.
+         options.level0_slowdown_writes_trigger = 20;     // number of L0 files that will slow down writes
+         options.level0_stop_writes_trigger = 40;         // number of L0 files that will stop writes
+         options.write_buffer_size = 256 * 1024 * 1024;   // memtable size
+         options.target_file_size_base = 256 * 1024 * 1024; // size of files in L1
+         options.max_bytes_for_level_base = 10 * options.target_file_size_base;  // total size of L1, recommended to be 10 * target_file_size_base
```

Comment:
This is a good idea. We should only provide options file capability. It will remove the confusion by keeping the previous rdb-max-files and rdb-threads.

Reply:
To avoid breaking existing test scripts, in the end, no-options-file is still allowed. But a warning is given.